torchtext/data/iterator.py

```diff
@@ -64,7 +65,7 @@ def __init__(self, dataset, batch_size, sort_key=None, device=None,
             logger.warning("The `device` argument should be set by using `torch.device`"
                            + " or passing a string as an argument. This behavior will be"
                            + " deprecated soon and currently defaults to cpu.")
-            device = None
+            device = torch.device('cpu')
```

Comment:
It makes sense that the default device is cpu if device is None. But if `device` is provided in the init func, you still force to use cpu. Seem not correct. And double check the doc explains this clearly.

Reply:
nit: could you remove the parentheses in Python conditional?